ipatests/test_integration/test_random_serial_numbers.py

```diff
@@ -64,3 +68,48 @@ def install(cls, mh):
         if not pki_supports_RSNv3(mh.master):
             raise pytest.skip("RNSv3 not supported")
         super(TestServerCALessToExternalCA_RSN, cls).install(mh)
+
+
+class TestStarTopo_RSN(IntegrationTest):
+    random_serial = True
+    topology = "star"
+
+    @classmethod
+    def install(cls, mh):
+        if not pki_supports_RSNv3(mh.master):
+            raise pytest.skip("RNSv3 not supported")
+        tasks.install_topo(
+            cls.topology, cls.master, cls.replicas, cls.clients
+        )
+
+
+class TestLineTopo_RSN(IntegrationTest):
+    random_serial = True
+    num_replicas = 2
+    topology = "line"
+
+    @classmethod
+    def install(cls, mh):
+        if not pki_supports_RSNv3(mh.master):
+            raise pytest.skip("RNSv3 not supported")
+        super(TestLineTopo_RSN, cls).install(mh)
+
+
+class TestRSNPKIConfig(IntegrationTest):
+    random_serial = True
+    num_replicas = 1
+
+    @classmethod
+    def install(cls, mh):
+        if not pki_supports_RSNv3(mh.master):
+            raise pytest.skip("RSNv3 not supported")
+        super(TestRSNPKIConfig, cls).install(mh)
+
+    def check_pki_config_params(self):
+        cs_cfg = self.master.get_file_contents(paths.CA_CS_CFG_PATH)
+        assert "dbs.cert.id.generator=random" in cs_cfg
+        assert "dbs.request.id.generator=random" in cs_cfg
+
+        cs_cfg = self.replicas[0].get_file_contents(paths.CA_CS_CFG_PATH)
+        assert "dbs.cert.id.generator=random" in cs_cfg
+        assert "dbs.request.id.generator=random" in cs_cfg
```

Comment:
FWIW when you do similar testing for the KRA the only difference is the first key which is dbs.key.id.generator.